__tests__/ExpensiMark-Markdown-Truncate-test.js

```diff
@@ -0,0 +1,101 @@
+    });
+
+    test('should handle HTML with nested markdown elements', () => {
+        const markdown = 'This is a *long _nested_ markdown* text. It contains *_bold italic_* and ~_strikethrough italic_~ formatting. The truncation should handle the nesting correctly.';
+        const limit = 150;
+        const html = parser.replace(markdown);
+        const result = parser.truncateHTML(html, limit, {ellipsis: true});
+        expect(result).toBe('This is a <strong>long <em>nested</em> markdown</strong> text. It contains <strong><em>bold italic</em></strong> and <del><em>strikethrough italic</em></del> formatting. The truncation should handle the nesting correctly.');
+    });
+
+    test('should handle HTML with links', () => {
+        const markdown =
+            'This is a text with [a link](https://example.com) that exceeds the limit. The link should be preserved in the truncated text. Additionally, it includes [another link](https://example.org) for testing purposes.';
+        const limit = 141;
+        const html = parser.replace(markdown);
+        const result = parser.truncateHTML(html, limit, {ellipsis: true});
+        expect(result).toBe(
+            `This is a text with <a href=\"https://example.com\" target=\"_blank\" rel=\"noreferrer noopener\">a link</a> that exceeds the limit. The link should be preserved in the truncated text. Additionally, it includes <a href=\"https://example.org\" target=\"_blank\" rel=\"noreferrer noopener\">another link</a>...`,
+        );
+    });
+
+    test('should handle HTML with inline code', () => {
+        const markdown = 'This is a text with `inline code`. The inline code should be preserved in the truncated text.';
+        const limit = 150;
+        const html = parser.replace(markdown);
+        const result = parser.truncateHTML(html, limit, {ellipsis: true});
+        expect(result).toBe('This is a text with <code>inline code</code>. The inline code should be preserved in the truncated text.');
+    });
+
+    test('should handle HTML with headings', () => {
+        const markdown =
+            '# Heading 1\n\nThis is a text with headings that exceeds the limit. The headings should be preserved in the truncated text. Here is an example of a heading:\n\n## Heading 2\n\nThe truncation should handle headings correctly.';
+        const limit = 100;
+        const html = parser.replace(markdown);
+        const result = parser.truncateHTML(html, limit, {ellipsis: true});
+        expect(result).toBe('<h1>Heading 1</h1><br />This is a text with headings that exceeds the limit. The headings should be preserved in th...');
+    });
+
+    test('should handle HTML with lists', () => {
+        const markdown = 'This is a text with lists. Here is an example of a list:\n\n- Item 1\n- Item 2\n- Item 3\n\nThe truncation should handle lists correctly.';
```

Comment:
Technically this isn't a list, right? Just a series of newlines? Should we be using `-` or a numbered list in the test here?